File: examples/example_mdfs.py
import mdfs
import numpy as np
import pandas as pd
import json

#| Reading data
madelon = pd.read_csv("data/madelon.50.csv", dtype=float)
dim = madelon.shape
data = madelon[[col for col in madelon.columns[:dim[1]-1]]][:dim[0]].to_numpy()
decision = madelon['Class'][:dim[0]].astype(np.intc).to_numpy() - 1

dimensions = 1

result = mdfs.run(data, decision, dimensions=dimensions, seed=0, discretizations=30, n_contrast=30)

contrast_variables = mdfs.gen_contrast_variables(data, n_contrast=30, seed=0)

class NumpyEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, np.integer):
            return int(obj)
        if isinstance(obj, np.floating):
            return float(obj)
        if isinstance(obj, np.ndarray):
            return obj.tolist()
        return json.JSONEncoder.default(self, obj)


with open("data/MDFS_Result_python_" + str(dimensions) + "D.json", "w") as f:
    f.write(json.dumps(result, indent=4, cls=NumpyEncoder))

# for contrast variable debugging:
# np.savetxt("data/test_contrast_variables.csv", contrast_variables.data, delimiter="\t")

pairs = mdfs.compute_tuples(data=data, decision=decision, seed=0, discretizations=30, return_matrix=True)

np.savetxt("data/pairs.csv", pairs.stats, delimiter="\t")

File: examples/example_mdfs.R
library(MDFS)

data <- read.csv("data/madelon.50.csv")
c <- read.table("data/test_contrast_variables.csv")

x <- data[, 1:50]
y <- data[, 51]

contrast.mask <- c(rep.int(F, ncol(x)), rep.int(T, ncol(c)))

result_1d <- ComputeMaxInfoGains(x, y, dimensions = 1, seed = 0, discretizations = 30, contrast_data = c)
result_1d_pv <- ComputePValue(
    c(result_1d$IG, attr(result_1d, "contrast_igs")),
    dimensions = 1, divisions = 1, one.dim.mode = "exp", contrast.mask = contrast.mask)

result_2d <- ComputeMaxInfoGains(x, y, dimensions = 2, seed = 0, discretizations = 30, contrast_data = c)
result_2d_pv <- ComputePValue(
    c(result_2d$IG, attr(result_2d, "contrast_igs")),
    dimensions = 2, divisions = 1, one.dim.mode = "exp", contrast.mask = contrast.mask)
